### PR TITLE
Update CAO/DAC RBAC

### PR DESCRIPTION
**Pull request by tahmmee**
Allows Operator and Admission controller to fetch and validate new Custom Resources
such as scopes and collections

### CODE REVIEW

File: charts/couchbase-operator/templates/operator-deployment.yaml [hunk 1]
```diff
@@ -201,6 +207,14 @@ rules:
...
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - get
+  - create
+  - update
 ---
 kind: {{ printf "%sBinding" .Values.couchbaseOperator.scope }}
 apiVersion: rbac.authorization.k8s.io/v1
```
Comment (patrick-stephens) [inline charts/couchbase-operator/templates/operator-deployment.yaml:220]: We should also add the prometheus annotations

**Review by patrick-stephens — APPROVED** [review]
Necessary pain but we really should look into just using the operator code/cao for this.

**Comment by tahmmee** [comment]
> Necessary pain but we really should look into just using the operator code/cao for this rather than having to duplicate changes each time - or make it mandatory that any changes in the operator code must be reflected with a helm PR immediately.

Very much a pain, I also wanted to reuse the generated cao yaml, but I realized I needed to modify cao itself in order have it generate just the RBAC portion, otherwise I need to cut out the deployment & secret section, rm namespace info.  Is that also what you were thinking?

**Comment by patrick-stephens** [comment]
> > Necessary pain but we really should look into just using the operator code/cao for this rather than having to duplicate changes each time - or make it mandatory that any changes in the operator code must be reflected with a helm PR immediately.
> 
> Very much a pain, I also wanted to reuse the generated cao yaml, but I realized I needed to modify cao itself in order have it generate just the RBAC portion, otherwise I need to cut out the deployment & secret section, rm namespace info.  Is that also what you were thinking?

That sounds like a reasonable thing, maybe an extra option to filter generation. Possibly useful in general

**Pull request merged**